### PR TITLE
Create rearranging_cars.py

### PR DESCRIPTION
**Pull request by ReemaJiries**
Assignment 6

### CODE REVIEW

File: rearranging_cars.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        empty_park_position = map_slot_cars[0]
+        new_park = new_parking[new_slot_position]
+        old_park_position = map_slot_cars[new_park]        
+        if new_slot_position != old_park_position:
+            #checking if there is a need for an extra move of the wanted park in case it is full
+            if new_slot_position != empty_park_position:
+                old_parking[empty_park_position] = old_parking[new_slot_position]
+                map_slot_cars[old_parking[new_slot_position]] = empty_park_position
+                steps.append((new_slot_position, empty_park_position))
+
```
Comment (joanna-) [inline rearranging_cars.py:29]: Here I think we should again check if the move is required. Please consider this example:
rearranging_cars([3, 1, 2, 0], [0, 1, 2, 3])
In this case we need only one move from 0 to 3 in order to get arrangement. However, your solution gives the answer:
[(0, 3), (3, 0)].

**Review by joanna- — CHANGES_REQUESTED** [review]
I think the code is simple and clean, easy to read. Good job! :) 

I pointed out one corner case in which your solution is failing though.

File: rearranging_cars.py (hunk 1 above)
Comment (ReemaJiries) [reply rearranging_cars.py:29]: Thank You for the review, I have correct it. I have to add more tests but I have exams right now so I will add it later.

**Review by joanna- — APPROVED** [review]

File: rearranging_cars.py (hunk 1 above)
Comment (joanna-) [reply rearranging_cars.py:29]: Thank you for the fix. Fingers crossed for your exams! :)

**Review by galeal2 — COMMENTED** [review]
Very nice! 😉

**Review by yuvdanilova — APPROVED** [review]
Your code is good!

I personally found it a bit hard visually as you prefer to longer variable names (it's great as it's understandable) and yet keep it all in one function. Maybe if you could split the code or, for example, shorten 'position' to 'pos', it would be a bit easier to read.

Though these are very minor issues and overall it's cool 👍

File: rearranging_cars.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                old_parking[empty_park_position] = old_parking[new_slot_position]
+                map_slot_cars[old_parking[new_slot_position]] = empty_park_position
+                steps.append((new_slot_position, empty_park_position))
+                
+            old_parking[new_slot_position] = new_park
+            old_parking[old_park_position] = 0
+            map_slot_cars[new_park] = new_slot_position
+            map_slot_cars[0] = old_park_position
+            steps.append((old_park_position, new_slot_position))
+
```
Comment (yuvdanilova) [inline rearranging_cars.py:35]:
```
    final_arrangement = []
    for i in range(len(old_parking)):
        final_arrangement.append(map_slot_cars[i])
    return final_arrangement
```

If I add this code here, should it return the right arrangement?
If yes, it fails on this input:

init: [6, 10, 2, 3, 8, 4, 5, 7, 1, 0, 9]
suggested output: [2, 1, 10, 8, 7, 3, 6, 9, 0, 4, 5]
actual output: [8, 1, 0, 5, 9, 10, 6, 4, 3, 7, 2]

Comment (ReemaJiries) [reply rearranging_cars.py:35]: You can just print(old_parking) and it's will output the actual output correctly, I have tried it and it passed

**Review by ariava — APPROVED** [review]
Good job. When you have time please add a top-level comment explaining the general strategy you applied to develop your solution.

**Pull request merged**